UI/output-obj.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "obs.hpp"
+
+enum class OutputType {
+	Unknown,
+	Preview,
+	Program,
+	Source,
+	Multiview, // TODO
+};
+
+class OutputObj {
+	bool active = false;
+
+	enum OutputType type;
+
+	OBSOutputAutoRelease output;
+	OBSWeakSourceAutoRelease weakSource;
```

Comment:
AFAIK, while views hold strong references to sources when they are set, they do not handle the `remove` signal. I think you need to handle that signal yourself to prevent removed sources from being held indefinitely.